src/Microsoft.ApplicationInsights.AspNetCore/Implementation/ApplicationInsightsInitializer.cs

```diff
@@ -41,6 +47,36 @@ internal class ApplicationInsightsInitializer : IObserver<DiagnosticListener>, I
         public void Start()
         {
             DiagnosticListener.AllListeners.Subscribe(this);
+
+            // TODO: Ideally we do not want to start this listener until we want to
+            var hostingEventSource = EventSource.GetSources()
+                .Where(eventSource => eventSource.Name == "Microsoft-AspNetCore-Hosting")
+                .FirstOrDefault();
+            if (hostingEventSource != null)
+            {
+                var args = new Dictionary<string, string>();
+                args.Add("EventCounterIntervalSec", "1");
+
+                // Sample EventCounter info
+                //< Event MSec = "4247.7123" PID = "20476" PName = "dotnet" TID = "23864" EventName = "EventCounters"
+                //  TimeStamp = "12/14/16 14:35:17.603350" ID = "27" Version = "0" Keywords = "0x00000000" TimeStampQPC = "2,376,979,690,831"
+                //  Level = "Always" ProviderName = "Microsoft-AspNetCore-Hosting" ProviderGuid = "9e620d2a-55d4-5ade-deb7-c26046d245a8" ClassicProvider = "False"
+                //  Opcode = "0" Task = "Default" Channel = "11" PointerSize = "8"
+                //  CPU = "2" EventIndex = "145728" TemplateType = "DynamicTraceEventData" >
+                //      < PrettyPrint >
+                //        < Event MSec = "4247.7123" PID = "20476" PName = "dotnet" TID = "23864" EventName = "EventCounters" ProviderName = "Microsoft-AspNetCore-Hosting" Payload = "{ Name=&quot;Request&quot;, Mean=NaN, StandardDerivation=NaN, Count=0, Min=0, Max=0, IntervalSec=1.000162 }" />
+                //               </ PrettyPrint >
+                //               < Payload Length = "40" >
+                //                     0:   e  0 52  0 65  0 71  0 | 75  0 65  0 73  0 74  0..R.e.q.u.e.s.t.
+                //      10:   0  0 c0 ff  0  0 c0 ff | 0  0  0  0  0  0  0  0........ ........
+                //      20:   0  0  0  0 4f  5 80 3f |                           ....O..?
+                //  </ Payload >
+                //</ Event >
+
+                //TODO: figure out when to disable listening for these events (Dispose does not seem right as this initializer instance is singleton
+                // which means it would remain as long as the app is up)
+                this.hostingEventSourceListener.EnableEvents(hostingEventSource, EventLevel.LogAlways, EventKeywords.None, args);
```

Comment:
@kichalla do you have documentation about it?

We just implemented counters aggregation in Application Insights SDK. It will be interesting to compare our implementation and this one. @vitalyf, can you please take a look?